### PR TITLE
Add Block limit upgrade

### PR DESCRIPTION
**Pull request by MertUnverdi**

### CODE REVIEW

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java [hunk 1]
```diff
@@ -33,13 +33,24 @@ public void onBlockPlaceEvent(BlockPlaceEvent event) {
             }
             return;
         }
-
         XMaterial material = XMaterial.matchXMaterial(event.getBlock().getType());
         if (!IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), user, PermissionType.BLOCK_PLACE)) {
             event.setCancelled(true);
-            player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotPlaceBlocks.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
+            player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotPlaceBlocks
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java:39]: I dont like this formatting

Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java:39]: oof

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java [hunk 2]
```diff
@@ -33,13 +33,24 @@ public void onBlockPlaceEvent(BlockPlaceEvent event) {
...
+                    .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
         }
+
         IslandBlocks islandBlocks = IridiumSkyblock.getInstance().getIslandManager().getIslandBlock(island.get(), material);
+        int limitUpgradeLevel = IridiumSkyblock.getInstance().getIslandManager().getIslandUpgrade(island.get(), "blocklimit").getLevel();
+        int blockLimit = IridiumSkyblock.getInstance().getUpgrades().blockLimitUpgrade.upgrades.get(limitUpgradeLevel).limits.get(material);
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java:45]: material might not be in this list

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java (hunk 1 above)
Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java:39]: others is fine?

**Review by PeachesMLG — CHANGES_REQUESTED** [review]
Changes above

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java (hunk 1 above)
Comment (PeachesMLG) [reply src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java:39]: I prefer things to be on one line unless there are alot of .replace's

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java [hunk 3]
```diff
@@ -33,13 +33,22 @@ public void onBlockPlaceEvent(BlockPlaceEvent event) {
...
-
         XMaterial material = XMaterial.matchXMaterial(event.getBlock().getType());
         if (!IridiumSkyblock.getInstance().getIslandManager().getIslandPermission(island.get(), user, PermissionType.BLOCK_PLACE)) {
             event.setCancelled(true);
             player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().cannotPlaceBlocks.replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix)));
         }
+
         IslandBlocks islandBlocks = IridiumSkyblock.getInstance().getIslandManager().getIslandBlock(island.get(), material);
+        int limitUpgradeLevel = IridiumSkyblock.getInstance().getIslandManager().getIslandUpgrade(island.get(), "blocklimit").getLevel();
+        int blockLimit = IridiumSkyblock.getInstance().getUpgrades().blockLimitUpgrade.upgrades.get(limitUpgradeLevel).limits.getOrDefault(material,0);
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java:44]:
```suggestion
        int blockLimit = IridiumSkyblock.getInstance().getUpgrades().blockLimitUpgrade.upgrades.get(limitUpgradeLevel).limits.getOrDefault(material, 0);
```

File: src/main/java/com/iridium/iridiumskyblock/configs/Upgrades.java [hunk 4]
```diff
@@ -55,6 +53,44 @@
             .put(3, new MemberUpgrade(1000, 15, 27))
             .build());
 
+    public Upgrade<BlockLimitUpgrade> blockLimitUpgrade = new Upgrade<>(true,
+            new Item(XMaterial.HOPPER, 13, 1, "&b&lIsland Block Limits", Arrays.asList(
+                    "&7Need to place more blocks? Buy this",
+                    "&7upgrade to increase the blocks you can place.",
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/configs/Upgrades.java:59]:
```suggestion
                    "&7upgrade to increase the amount of blocks you can place.",
```

File: src/main/java/com/iridium/iridiumskyblock/configs/Messages.java [hunk 5]
```diff
@@ -130,6 +130,7 @@
     public String bankDeposited = "%prefix% &7You successfully deposited %amount% %type% from your Island bank.";
     public String insufficientFundsToWithdrew = "%prefix% &7You do not have enough %type% to withdraw from your Island bank.";
     public String insufficientFundsToDeposit = "%prefix% &7You do not have enough %type% to deposit into your Island bank.";
+    public String blockLimitReached = "%prefix% &7The maximum block limit for %block%(%limit%) is reached!";
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/configs/Messages.java:133]:
```suggestion
    public String blockLimitReached = "%prefix% &7The maximum block limit for %block% (%limit%) is reached!";
```

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java [hunk 6]
```diff
@@ -33,13 +33,22 @@ public void onBlockPlaceEvent(BlockPlaceEvent event) {
...
+
+        if (blockLimit > 0 && islandBlocks.getAmount() >= blockLimit) {
+            player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().blockLimitReached
+                    .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix).replace("%limit%", String.valueOf(blockLimit)).replace("%block%", material.name())));
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java:48]: You should use something better than Material#name(), enum constant names don't look that good. Maybe replace the _ with spaces, make it lowercase and capitalize it

**Review by dlsf — CHANGES_REQUESTED** [review]

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java (hunk 3 above)
Comment (MertUnverdi) [reply src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java:44]: damn

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java (hunk 6 above)
Comment (PeachesMLG) [reply src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java]: Check out what we did with Biome GUI

File: src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java [hunk 7]
```diff
@@ -41,11 +41,11 @@ public void onBlockPlaceEvent(BlockPlaceEvent event) {
 
         IslandBlocks islandBlocks = IridiumSkyblock.getInstance().getIslandManager().getIslandBlock(island.get(), material);
         int limitUpgradeLevel = IridiumSkyblock.getInstance().getIslandManager().getIslandUpgrade(island.get(), "blocklimit").getLevel();
-        int blockLimit = IridiumSkyblock.getInstance().getUpgrades().blockLimitUpgrade.upgrades.get(limitUpgradeLevel).limits.getOrDefault(material,0);
+        int blockLimit = IridiumSkyblock.getInstance().getUpgrades().blockLimitUpgrade.upgrades.get(limitUpgradeLevel).limits.getOrDefault(material, 0);
 
         if (blockLimit > 0 && islandBlocks.getAmount() >= blockLimit) {
             player.sendMessage(StringUtils.color(IridiumSkyblock.getInstance().getMessages().blockLimitReached
-                    .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix).replace("%limit%", String.valueOf(blockLimit)).replace("%block%", material.name())));
+                    .replace("%prefix%", IridiumSkyblock.getInstance().getConfiguration().prefix).replace("%limit%", String.valueOf(blockLimit)).replace("%block%", material.name().replace("_", " "))));
```
Comment (PeachesMLG) [inline src/main/java/com/iridium/iridiumskyblock/listeners/BlockPlaceListener.java:48]: Check out what we did with biome GUI

**Review by PeachesMLG — APPROVED** [review]
@dlsf this look good to u?

**Comment by dlsf** [comment]
3/4 comments of my previous review are still not resolved 🤔

**Comment by PeachesMLG** [comment]
r u sure? looks like they are resolved to me?

File: src/main/java/com/iridium/iridiumskyblock/configs/Messages.java [hunk 8]
```diff
@@ -130,6 +130,7 @@
     public String bankDeposited = "%prefix% &7You successfully deposited %amount% %type% from your Island bank.";
     public String insufficientFundsToWithdrew = "%prefix% &7You do not have enough %type% to withdraw from your Island bank.";
     public String insufficientFundsToDeposit = "%prefix% &7You do not have enough %type% to deposit into your Island bank.";
+    public String blockLimitReached = "%prefix% &7The maximum block limit for %block% (%limit%) is reached!";
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/configs/Messages.java:133]:
```suggestion
    public String blockLimitReached = "%prefix% &7The maximum block limit for %block% (%limit%) has been reached!";
```

File: src/main/java/com/iridium/iridiumskyblock/configs/Upgrades.java [hunk 9]
```diff
@@ -55,6 +53,44 @@
             .put(3, new MemberUpgrade(1000, 15, 27))
             .build());
 
+    public Upgrade<BlockLimitUpgrade> blockLimitUpgrade = new Upgrade<>(true,
+            new Item(XMaterial.HOPPER, 13, 1, "&b&lIsland Block Limits", Arrays.asList(
+                    "&7Need to place more blocks? Buy this",
+                    "&7upgrade to increase the amount of  blocks you can place.",
```
Comment (dlsf) [inline src/main/java/com/iridium/iridiumskyblock/configs/Upgrades.java:59]:
```suggestion
                    "&7upgrade to increase the amount of blocks you can place.",
```

**Review by dlsf — CHANGES_REQUESTED** [review]
You are right

**Pull request merged**